airbyte-integrations/connectors/source-slack/source_slack/spec.json

```diff
@@ -31,7 +30,84 @@
         "default": true,
         "title": "Join all channels",
         "description": "Whether to join all channels or to sync data only from channels the bot is already in.  If false, you'll need to manually add the bot to all the channels from which you'd like to sync messages. "
+      },
+      "credentials": {
+        "title": "Authentication mechanism",
+        "description": "Choose either to provide the API key or the OAuth2.0 credentials",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "OAuth2.0 authorization",
+            "required": [
+              "access_token",
+              "client_id",
+              "client_secret",
+              "option_title"
+            ],
+            "properties": {
+              "option_title": {
+                "type": "string",
+                "const": "Default OAuth2.0 authorization"
+              },
+              "client_id": {
+                "title": "Client ID",
+                "description": "Slack client_id. See our <a href=\"https://docs.airbyte.io/integrations/sources/slack\">docs</a> if you need help finding this id.",
+                "type": "string",
+                "examples": ["slack-client-id-example"]
+              },
+              "client_secret": {
+                "title": "Client Secret",
+                "description": "Slack client_secret. See our <a href=\"https://docs.airbyte.io/integrations/sources/slack\">docs</a> if you need help finding this secret.",
+                "type": "string",
+                "examples": ["slack-client-secret-example"],
+                "airbyte_secret": true
+              },
+              "access_token": {
```

Comment:
i suggest we remove the access token since it expires anyways, and just rely on refresh token

Reply:
comment above